File: SWEA/1210.py
def findtwo(nums):
    for i in range(1,101):
        if nums[99][i] == 2:
            return i


for i in range(10):
    n = int(input())
    start = 0
    
    m = 99
    nums = [[0 for x in range(102)] for y in range(100)]

    for j in range(100):
        nums[j][1:101] = list(map(int, list(input().split())))

    start = findtwo(nums)

    while m > 0:
        
        if nums[m][start-1]:
            while nums[m][start-1]:
                start -= 1
            m -= 1
            continue
        
        elif nums[m][start+1]:
            while nums[m][start+1]:
                start += 1
            m -= 1
            continue
            
        else :
            m -= 1
        
    print(f'#{n} {start-1}')

File: SWEA/1210.cpp
#include <iostream>

using namespace std;

int main() {
    int start, n = 0;
    
    int nums[100][100];
    for (int i=0; i<10; ++i) {
        int m = 99;
        cin >> n;
        

        for (int j = 0; j < 100; ++j)
            for (int k = 0; k < 100; ++k)
                cin >> nums[j][k];

        for (int c=0;c<100;++c) {
            if (nums[99][c] == 2) {
                start = c;
                break;
            }
        }

        while (m > 0) {
            if (start >0 and nums[m][start-1] == 1) {
                while (start >0 and nums[m][start-1] == 1) {
                    --start;
                    
                }
                --m;
                continue;
            } else if (start<99 and nums[m][start+1] == 1) {
                while (start<99 and nums[m][start+1] == 1) {
                    ++start;
                    
                }
                --m;
                continue;
            } else {
                --m;
            }
        }

            
        
        cout << "#" << i+1 << " " << start << endl;
    }
}